### PR TITLE
AP-8

### PR DESCRIPTION
**Pull request by rdhawladar**

### CODE REVIEW

File: internal/app/parcel/repository.go [hunk 1]
```diff
@@ -30,23 +30,25 @@ func NewRepository(db *sqlx.DB) *repository {
...
+
+	err = stmt.GetContext(ctx, &parcel, &parcel)
+
+	if err != nil {
 		if pqErr, ok := err.(*pq.Error); ok && pqErr.Code == errUniqueViolation {
-			return fmt.Errorf("%v :%w", err, model.ErrInvalid)
+			return model.Parcel{}, fmt.Errorf("%v :%w", err, model.ErrInvalid)
 		}
-		return err
+		log.Error().Err(err).Msgf("[FetchParcelByID] Error from getcontext 2: %v", err)
```
Comment (shuza) [inline internal/app/parcel/repository.go]: use proper log message. That's I was tell about not to use this message 😂

File: internal/app/parcel/repository_test.go [hunk 2]
```diff
@@ -33,41 +34,61 @@ func TestRepository_InsertParcel(t *testing.T) {
...
+					parcel.CompanyFee,
+					parcel.CreatedAt,
+					parcel.UpdatedAt))
 
 		repo := NewRepository(sqlxDB)
-		err := repo.InsertParcel(context.Background(), parcel)
+		result, err := repo.InsertParcel(context.Background(), parcel)
+
 		assert.Nil(t, err)
+		assert.Equal(t, result, parcel)
```
Comment (shuza) [inline internal/app/parcel/repository_test.go:65]: Did you check difference between `Equal` and. `EqualValues`?

File: internal/app/parcel/service_test.go [hunk 3]
```diff
@@ -18,10 +18,11 @@ var parcel = model.Parcel{
 	DestinationAddress: "Pabna Shadar",
 	SourceTime:         time.Now(),
 	ParcelType:         "Document",
-	Status:             2,
 	Price:              200.0,
 	CarrierFee:         180.0,
 	CompanyFee:         20.0,
+	CreatedAt:          time.Date(2020, time.April, 11, 21, 34, 01, 0, time.UTC),
```
Comment (shuza) [inline internal/app/parcel/service_test.go:24]: @rdhawladar good job 👍

File: internal/app/parcel/service_test.go [hunk 4]
```diff
@@ -57,7 +58,7 @@ func TestService_CreateParcel(t *testing.T) {
 	for _, tc := range testCases {
 		t.Run(tc.desc, func(t *testing.T) {
 			s := NewService(tc.mockRepo())
-			err := s.CreateParcel(context.Background(), parcel)
+			_, err := s.CreateParcel(context.Background(), parcel)
```
Comment (shuza) [inline internal/app/parcel/service_test.go]: we should also check the value returned by `CreateParcel` method

**Review by shuza — CHANGES_REQUESTED** [review]
update log message

File: internal/app/parcel/repository_test.go (hunk 2 above)
Comment (rdhawladar) [reply internal/app/parcel/repository_test.go:65]: Yes, equal  checks the type also. If type is same, should use equal.

**Review by shuza — APPROVED** [review]

**Pull request merged**